### DISCUSSION TITLE
Logging is being applied to everything instead of just gunicorn/uvicorn

### DISCUSSION THREAD

**Discussion by jeanlst** [br3ndonland/inboard#48] [Q&A] [answered]
I'm not completely aware of the flow here but I'm using Fastapi+Uvicorn+Gunicorn with inboard. However, every log from gunicorn/my app/uvicorn is being duplicated, what's the purpose of having `propagate` in every logger?

**Comment by br3ndonland** [comment, marked as answer]
inboard propagates FastAPI, Gunicorn, and Uvicorn log streams to the root logger, so that all the log streams can be handled by the same logging config.

If you don't like inboard's logging behavior, you can modify it, or override it completely. See the [logging docs](https://inboard.bws.bio/logging) for more info and examples.

If you are seeing duplicate log messages, there may be something project-specific that is contributing. Do you have any other loggers or logging configs in the project?
